sql/telemetry_derived/clients_histogram_aggregates_v1/query.sql

```diff
@@ -143,23 +147,93 @@ normalized_histograms AS
       metric_type,
       key,
       agg_type,
-      latest_version)
+      latest_version),
 
-SELECT
+clients_histogram_aggregates_new AS
+  (SELECT
     client_id,
     os,
     app_version,
     app_build_id,
     channel,
-    first_bucket,
-    last_bucket,
-    num_buckets,
-    metric,
-    metric_type,
-    normalized_histograms.key AS key,
-    agg_type,
-    agg.key AS bucket,
-    agg.value AS value
-FROM normalized_histograms
-CROSS JOIN UNNEST(aggregates) AS agg
-WHERE num_buckets > 0
+    ARRAY_AGG(STRUCT<
+      first_bucket INT64,
+      last_bucket INT64,
+      num_buckets INT64,
+      latest_version INT64,
+      metric STRING,
+      metric_type STRING,
+      key STRING,
+      agg_type STRING,
+      aggregates ARRAY<STRUCT<key STRING, value INT64>>>(
+        first_bucket,
+        last_bucket,
+        num_buckets,
+        latest_version,
+        metric,
+        metric_type,
+        key,
+        agg_type,
+        aggregates
+  )) AS histogram_aggregates
+  FROM aggregated_histograms
+  GROUP BY
+    client_id,
+    os,
+    app_version,
+    app_build_id,
+    channel),
+
+clients_histogram_aggregates_old AS
+  (SELECT
+    client_id,
+    os,
+    app_version,
+    app_build_id,
+    histogram_aggs.channel AS channel,
+    histogram_aggregates
+  FROM clients_histogram_aggregates_v1 AS histogram_aggs
+  LEFT JOIN latest_versions
+  ON latest_versions.channel = histogram_aggs.channel
+  WHERE app_version >= (latest_version - 2)),
+
+joined_new_old AS (
+  SELECT
+    CASE
+      WHEN old_data.client_id IS NOT NULL THEN old_data.client_id
+      ELSE new_data.client_id
+    END AS client_id,
```

Comment:
This can be conveyed more concisely as `COALESCE(old_data.client_id, new_data.client_id) AS client_id`, same for the following case statements here.